File: Go/Easy/Palindrome Check/solution_1.go
package main

func IsPalindrome(str string) bool {

	left := 0
	right := len(str) - 1
	isFound := false

	for left <= right {

		if str[left] == str[right] {
			isFound = true
			left += 1
			right -= 1
		} else {
			isFound = false
			return isFound
		}
	}

	return isFound
}

File: Python/Easy/Palindrome Check/solution_1.py
def isPalindrome(string):
    
    left = 0
    right = len(string) - 1
    isFound = False

    while left <= right:

        if string[left] == string[right]:
            isFound = True
            left += 1
            right -= 1
        else:
            isFound = False
            return isFound

    return isFound